WPT webrtc/RTCRtpTransceiver-setCodecPreferences.html last test is buggy in case of multiple codec variants

youennf opened the pull request:
#### f987e4ff207694211ab0c2d42c01e2728c2a2443
<pre>
WPT webrtc/RTCRtpTransceiver-setCodecPreferences.html last test is buggy in case of multiple codec variants
<a href="https://bugs.webkit.org/show_bug.cgi?id=243338">https://bugs.webkit.org/show_bug.cgi?id=243338</a>

Reviewed by Eric Carlson.

We should only swap the first H264 with the first VP8 entries.

* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCRtpTransceiver-setCodecPreferences-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCRtpTransceiver-setCodecPreferences.html:

Canonical link: <a href="https://commits.webkit.org/252952@main">https://commits.webkit.org/252952@main</a>
</pre>

eric-carlson commented on Source/WebCore/Modules/mediastream/RTCPeerConnection.cpp:
```diff
@@ -577,6 +577,9 @@ bool RTCPeerConnection::doClose()
 
+    for (auto& transport : m_dtlsTransports)
+        transport->close();
+
```
Looks like you have two changes in this PR

youennf replied:
Gasp, I pushed too eagerly!, will fix this shortly

eric-carlson reviewed (approved):
r=me

youennf committed:
WPT webrtc/RTCRtpTransceiver-setCodecPreferences.html last test is buggy in case of multiple codec variants
https://bugs.webkit.org/show_bug.cgi?id=243338

Reviewed by Eric Carlson.

We should only swap the first H264 with the first VP8 entries.

* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCRtpTransceiver-setCodecPreferences-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/webrtc/RTCRtpTransceiver-setCodecPreferences.html:

Canonical link: https://commits.webkit.org/252952@main
